sql/base_session.go

```diff
@@ -147,7 +147,7 @@ func (s *BaseSession) SetSessionVariable(ctx *Context, sysVarName string, value
 		}
 	}
 
-	if !sysVar.Var.Dynamic {
+	if !sysVar.Var.Dynamic || sysVar.Var.ValueFunction != nil {
```

Comment:
I looked at adding an error if `Dyanmic` is set to `true` when a `ValueFunction` is provided, but there wasn't a good place to return an error (e.g. sys variables are often loaded in an `init` function)